### DISCUSSION TITLE
Add session type in public schedule

### DISCUSSION THREAD

**Discussion by sparunakian** [pretalx/pretalx#1287] [Ideas] [answered]
Hi,

Public schedule doesn't show session type, in full view and in talk view. Sometimes the session type matters.

In our use case (https://pretalx.jdll.org), we have a session type dedicated for activities for children. So it would be quite useful to display the session type, to prevent non-children coming to children activities :-)

Regards,

**Comment by rixx** [comment, marked as answer]
Session types are currently primarily meant to help organisers juggle default durations and filter talks. We don't really have a good way to show yet another way of categorising talks in the schedule, and since colours are the easiest thing to filter for at a glance, that's usually what tracks are used for.

**Comment by sparunakian** [comment]
So I've just tested tracks on our last event and we now have a way to differentiate talks in the public schedule, thank you. But I had to mark every talk with newly created tracks. Is there a way to link a session type with a track? Or to restrict some tracks to a session type, to let speakers select themselves? I've searched Pretalx and the issues, in vain.

**Reply by rixx** [reply]
There is no way to do that without resorting to the Python shell, no – tracks are intended to be added at the time of submission, so there has never been a need to change tracks in bulk.

**Reply by sparunakian** [reply]
Is there a way to restrict tracks to a session type in the cfp? Or to set a default by session type? Maye I've misunderstood something, because we created tracks identical to session types so that the planning has colors.
In my example with activities for children, we have a dedicated duration (specified in a dedicated session type). We could let speakers choose a track, but if they have chosen the "Children" session type, there is only one track matching. We then have to manually verify that the track they chose (if they chose one) is correct.

**Reply by rixx** [reply]
There's currently no way to a specific session type, no, and it would be fairly hard to introduce that functionality, since we'd suddenly need to teach the frontend about backend business logic, which we have avoided so far. Having tracks tightly coupled to session types really hasn't come up so far, so there is no tooling for any of this (though we are working on providing a bulk change interfaces – hopefully this year? No promises though.)

**Reply by sparunakian** [reply]
Okay, thank you for your response!
